### PR TITLE
feat(TDI-39298): GoogleDrive Integration

### PR DESCRIPTION
**Pull request by undx**
**What is the current behavior?** (You can also link to an open issue here)

https://jira.talendforge.org/browse/TDI-39298

Initial PR https://github.com/Talend/components/pull/998 validated till datastore and dataset commits.

**What is the new behavior?**

**Please check if the PR fulfills these requirements**

- [X] The commit message follows Talend standard
- [X] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features) ?
- [X] The code coverage on new code >75%
- [ ] The new code does not introduce new technical issues (sonar / eslint)

**What kind of change does this PR introduce?**

- [ ] Bugfix
- [X] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build / CI related changes
- [ ] Other... Please describe:

**Does this PR introduce a breaking change?**

- [ ] Yes
- [X] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

Code Coverage:
- definition: 98%
- runtime: 88%

### CODE REVIEW

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1466/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1466/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39298_GoogleDrive)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1488/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1488/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39298_GoogleDrive)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1494/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1494/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39298_GoogleDrive)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1542/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1542/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39298_GoogleDrive)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1548/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1548/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39298_GoogleDrive)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1550/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1550/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39298_GoogleDrive)

File: components/components-google-drive/components-google-drive-definition/src/main/java/org/talend/components/google/drive/data/GoogleDriveDatastoreProperties.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+            LOG.warn((messages.getMessage("error.env.not.set")));
+            saPath = this.getClass().getClassLoader().getResource("service_account.json").getFile();
+        }
+        serviceAccountJSONFile.setValue(saPath);
+    }
+
+    @Override
+    public void setupLayout() {
+        Form mainForm = Form.create(this, Form.MAIN);
+        // mainForm.addRow(serviceAccountJSONFile);
```
Comment (ypiel-talend) [inline components/components-google-drive/components-google-drive-definition/src/main/java/org/talend/components/google/drive/data/GoogleDriveDatastoreProperties.java]: Delete commented code

File: components/components-google-drive/components-google-drive-definition/src/main/java/org/talend/components/google/drive/data/GoogleDriveDatastoreDefinition.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+    }
+
+    @Override
+    public String getInputCompDefinitionName() {
+        return GoogleDriveInputDefinition.NAME;
+    }
+
+    @Override
+    public String getOutputCompDefinitionName() {
+        // none
```
Comment (ypiel-talend) [inline components/components-google-drive/components-google-drive-definition/src/main/java/org/talend/components/google/drive/data/GoogleDriveDatastoreDefinition.java]: Useless comment

File: components/components-google-drive/components-google-drive-definition/src/main/java/org/talend/components/google/drive/data/GoogleDriveDatastoreDefinition.java [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        return NAME + "_icon32.png";
+    }
+
+    @Override
+    public String getImagePath(DefinitionImageType definitionImageType) {
+        switch (definitionImageType) {
+        case PALETTE_ICON_32X32:
+            return NAME + "_icon32.png";
+        case SVG_ICON:
+            return null;
```
Comment (ypiel-talend) [inline components/components-google-drive/components-google-drive-definition/src/main/java/org/talend/components/google/drive/data/GoogleDriveDatastoreDefinition.java]: There is no generic default icon ?

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/1577/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/1577/)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1591/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1591/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39298_GoogleDrive)

**Comment by build-talend-common** [comment]
Pull request test result: [success](https://ci-common.datapwn.com/job/components_pull-request_master/1593/) [:star:](https://ci-common.datapwn.com/job/components_pull-request_master/1593/) 
[Sonar results](https://sonar.datapwn.com/overview?id=org.talend.components%3Acomponents%3Aundx%2FTDI-39298_GoogleDrive)

**Pull request merged**